openslides/agenda/projector.py

```diff
@@ -69,31 +69,63 @@ def get_collection_elements_required_for_this(self, collection_element, config_e
     """
-    name = 'agenda/current-list-of-speakers'
+    Main class for the list of speaker slides.
 
+    """
     def get_requirements(self, config_entry):
-        pk = config['projector_currentListOfSpeakers_reference']
-        if pk is not None:
-            # List of speakers slide.
-            try:
-                item = Item.objects.get(pk=pk)
-            except Item.DoesNotExist:
-                # Item does not exist. Just do nothing.
-                pass
-            else:
-                yield item
-                for speaker in item.speakers.filter(end_time=None):
-                    # Yield current speaker and next speakers
-                    yield speaker.user
+        items = self.get_agenda_items(config['projector_currentListOfSpeakers_reference'])
+        for item in items:
+            yield item
+            for speaker in item.speakers.filter(end_time=None):
+                yield speaker.user
                 query = (item.speakers.exclude(end_time=None)
                          .order_by('-end_time')[:config['agenda_show_last_speakers']])
                 for speaker in query:
                     # Yield last speakers
                     yield speaker.user
+
+    def get_agenda_items(self, projector_id):
+        projector = Projector.objects.get(pk=projector_id)
+        for element_searched in projector.elements:
+            if 'agenda_item_id' in projector.elements[element_searched]:
+                agenda_item_id = projector.elements[element_searched]['agenda_item_id']
+                # TODO: check in which cases agenda_item_id exist, but is ''
+                if agenda_item_id != '':
+                    yield Item.objects.get(pk=agenda_item_id)
+
+    def get_collection_elements_required_for_this(self, collection_element, config_entry):
+        output = super().get_collection_elements_required_for_this(collection_element, config_entry)
+        # Full update if agenda_item changes because then we may have new
+        # candidates and therefor need new users.
+        items = self.get_agenda_items(config['projector_currentListOfSpeakers_reference'])
+        for item in items:
+            requirements_as_collection_elements = list(self.get_requirements_as_collection_elements(item))
+            for collection_element in requirements_as_collection_elements:
+                if collection_element == CollectionElement.from_values(Item.get_collection_string(), item.id):
+                    output.extend(self.get_requirements_as_collection_elements(item))
+        return output
+
+
+class CurrentListOfSpeakersSlide(CurrentListOfSpeakersMetaClass):
+    """
+    Slide for the current list of speakers.
+    """
+    name = 'agenda/current-list-of-speakers'
+
+
+class CurrentListOfSpeakersOverlaySlide(CurrentListOfSpeakersMetaClass):
+    """
+    List of speakers overlay.
+    Subclass of ListOfSpeakers
+    """
+    name = 'agenda/current-list-of-speakers-overlay'
+    stable = True
```

Comment:
This is not necessary. The client has to write the flag `stable` into the projector config. Please remove this line.